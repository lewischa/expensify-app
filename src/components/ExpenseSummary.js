import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    const expenseWord = props.numExpenses !== 1 ? 'expenses ' : 'expense ';
    const formattedTotal = numeral(props.total / 100).format('$0,0.00');

    return (
        <div>
            Viewing {props.numExpenses} {expenseWord}
            totalling {formattedTotal}
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectVisibleExpenses(state.expenses, state.filters);
    return {
        numExpenses: visibleExpenses.length,
        total: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
