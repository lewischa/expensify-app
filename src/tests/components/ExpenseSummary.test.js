import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('ExpenseSummary snapshot should match with 0 expenses', () => {
    const wrapper = shallow(<ExpenseSummary numExpenses={0} total={0}/>);
    expect(wrapper).toMatchSnapshot();
});

test('ExpenseSummary snapshot should match with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary numExpenses={1} total={2346} />);
    expect(wrapper).toMatchSnapshot();
});

test('ExpenseSummary snapshot should match with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary numExpenses={23} total={2364236643} />);
    expect(wrapper).toMatchSnapshot();
});
