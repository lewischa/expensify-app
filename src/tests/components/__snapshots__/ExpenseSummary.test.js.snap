// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary snapshot should match with 0 expenses 1`] = `
<div>
  Viewing 
  0
   
  expenses 
  totalling 
  $0.00
</div>
`;

exports[`ExpenseSummary snapshot should match with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense 
  totalling 
  $23.46
</div>
`;

exports[`ExpenseSummary snapshot should match with multiple expenses 1`] = `
<div>
  Viewing 
  23
   
  expenses 
  totalling 
  $23,642,366.43
</div>
`;
